### ISSUE TITLE
BUG: Problem with "Registering source file signalp-3.0.Linux.tar.Z"

### ISSUE THREAD

**Issue by a-velt** [ccdmb/predector#70]
**Describe the bug**

The registration for signalp-3.0 failed with the command : 

`signalp3-register signalp-3.0.Linux.tar.Z`

**To Reproduce**

I have recovered the sources of all the proprietary tools. I installed predector with conda, so I have a dedicated conda environment. I had errors to register the proprietary tools, by adding the path to the libraries in LD_LIBRARY_PATH, that solved the problem for all except one : signalp-3.0. 

Just a note, I run "install.sh" to create the predictor environement, then I activate this environment, I added the "predector" environment's library path to LD_LIBRARY_PATH and THEN I launched the registration manually.

Here my commands : 

```
ENVIRONMENT=conda

cd /cm/shared/apps/predector/proprietary_software_dependencies

bash install.sh "${ENVIRONMENT}" \
-3 signalp-3.0.Linux.tar.Z \
-4 signalp-4.1g.Linux.tar.gz \
-5 signalp-5.0b.Linux.tar.gz \
-6 signalp-6.0d.fast.tar.gz \
-t targetp-2.0.Linux.tar.gz \
-d deeploc-1.0.All.tar.gz \
-m tmhmm-2.0c.Linux.tar.gz \
-p phobius101_linux.tar.gz

# the registrations failed
# I activate the predector environment and then I launched the registrations manually

# >>> conda initialize >>>
# !! Contents within this block are managed by 'conda init' !!
__conda_setup="$('/cm/shared/apps/predector/bin/conda' 'shell.bash' 'hook' 2> /dev/null)"
if [ $? -eq 0 ]; then
    eval "$__conda_setup"
else
    if [ -f "/cm/shared/apps/predector/etc/profile.d/conda.sh" ]; then
        . "/cm/shared/apps/predector/etc/profile.d/conda.sh"
    else
        export PATH="/cm/shared/apps/predector/bin:$PATH"
    fi
fi
unset __conda_setup
# <<< conda initialize <<<
export LD_LIBRARY_PATH=/cm/shared/apps/predector/envs/predector/lib/:$LD_LIBRARY_PATH

conda activate predector

signalp3-register signalp-3.0.Linux.tar.Z
signalp4-register signalp-4.1g.Linux.tar.gz
signalp5-register signalp-5.0b.Linux.tar.gz
signalp6-register signalp-6.0d.fast.tar.gz
targetp2-register targetp-2.0.Linux.tar.gz
deeploc-register deeploc-1.0.All.tar.gz
phobius-register phobius101_linux.tar.gz
tmhmm2-register tmhmm-2.0c.Linux.tar.gz
```

Registrations worked for all the tools except signalp-3.0

`signalp3-register signalp-3.0.Linux.tar.Z`

I have this error : 

```
Registering source file signalp-3.0.Linux.tar.Z for signalp3 into conda environment at:
/cm/shared/apps/predector/envs/predector/share/signalp3-3.0b-1

Unregistering old source files if they exist.

tar (child): compress: Cannot exec: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
basename: missing operand
Try 'basename --help' for more information.
```

EDIT : 

In the "/cm/shared/apps/predector/envs/predector/share/signalp3-3.0b-1/register.sh", if I lanuch : 

`tar -tf "signalp-3.0.Linux.tar.Z" | head -n 1`

I have the error : 

```
tar (child): compress: Cannot exec: No such file or directory
tar (child): Error is not recoverable: exiting now
tar: Child returned status 2
tar: Error is not recoverable: exiting now
```

There is a problem with the tar command for sure. I downloaded signalp-3.0.Linux.tar.Z twice, to be sure there is no problem with my first download, but I got the same error.

**Expected behavior**
I expect the message "signalp-3.0 is now full installed!"

**Screenshots or copy-pasted terminal outut**
/

**Operating system (please enter the following information as appropriate):**
 - CentOS
 - Conda
 - Are you using MacOS or Windows WSL?    NO

**Additional context**
/

**Comment by darcyabjones** [comment]
Hi @a-velt !

Thanks for reporting the issue.
I'll try to find a fix for this.
Given that it's the only one you're having trouble with, my suspicion is that it will have something to do with tar not being able to detect/handle the older `.Z` compression.
Possibly your centos distro is distributing a different version of tar to what I usually test with.

Can I please get some extra details from you, so that I know what I need to handle?
- Which version of centos are you running? `cat /etc/centos-release`
- Which bash version are you running? `bash --version`
- Which tar version are you running? `tar --version`
- Is "ncompress" installed? If no, try running `yum install ncompress` and run the register script again (it might be `dnf` instead of `yum` now, depends on what version you're running).

In the short term, if docker or singularity is an option for you I'd highly recommend it because the environment is consistent.
Otherwise, you can run the commands in the signalp3 register file yourself.
Its really only the untar-ing command that's failing, so if you can get that uncompressed the rest should be ok.
Something like below should work, but it's late over here so I haven't tested it and I'd need the versions.

```
# uncompress the archive yourself
# using Z should force it to recognise the correct uncompression program.
# Lower case z might also work (i.e. using gzip)
tar --no-same-owner -Zxf signalp-3.0.Linux.tar.Z

# If the above doesn't work, try uncompressing separately
uncompress signalp-3.0.Linux.tar.Z
# gunzip signalp-3.0.Linux.tar.Z   may also work
tar --no-same-owner -xf signalp-3.0.Linux.tar


cd  # the extracted directory


TARGET_DIR="/cm/shared/apps/predector/envs/predector/share/signalp3-3.0b-1"

cp -r ./* "${TARGET_DIR}"
cd "${TARGET_DIR}"

patch signalp signalp.patch
sed -i 's~`which nawk`~gawk~' bin/testhow

chmod -R a+r .
chmod a+x signalp
chmod a+x signalp bin/* how/* hmm/*

# Check that it runs without issue.
signalp3 -t euk -short "${TARGET_DIR}/test/test5.seq"
```

Hope that gets you going in the short term.

Thanks again

**Comment by a-velt** [comment]
Hi,

I installed "ncompress" and now 

`signalp3-register signalp-3.0.Linux.tar.Z`

works !

```
Registering source file signalp-3.0.Linux.tar.Z for signalp3 into conda environment at:
/cm/shared/apps/predector/envs/predector/share/signalp3-3.0b-1

Unregistering old source files if they exist.

patching file signalp
Finished registering signalp3.
Testing installation...
Test succeeded.
signalp3 is now fully installed!
```

**Comment by darcyabjones** [comment]
Wonderful!

That was really kind of a wild guess, i'm amazed it was that easy!
Thanks again for reporting the issue and getting back to me, it really does help a lot.

I'll see if I can handle the issue automatically in the installer.
I'll close this issue when that's done.

Cheers,
Darcy

**Comment by darcyabjones** [comment]
Hey Amandine,

I wasn't able to find any centos distros that I could reproduce your issue with. Everything seems to come with `ncompress` installed by default.
`gunzip` is able to handle files compressed with `compress` (the old algorithm used to create `.Z` files).
So this should at least be more portable.

I've updated to specify the `-z` flag to `tar`, and I've made `gzip` a dependency of the package.
I'm hoping that avoids the issue for the future, but without a way to reproduce the issue it's hard for me to do more.

Fixed with 6d2b6a16a9c87a4d86f5b3dbcfbd1ab7b2be8e48 to be introduced in upcoming patch.

Thanks again for reporting the issue!

Darcy
